src/app-layer-htp.c

```diff
@@ -2157,7 +2175,23 @@ static int HTPCallbackDoubleDecodePath(htp_tx_t *tx)
         return HTP_OK;
 
     uint64_t flags = 0;
-    htp_urldecode_inplace(tx->cfg, HTP_DECODER_URL_PATH, tx->parsed_uri->path, &flags);
+    size_t prevlen = bstr_len(tx->parsed_uri->path);
+    htp_status_t res = htp_urldecode_inplace(tx->cfg, HTP_DECODER_URL_PATH, tx->parsed_uri->path, &flags);
+    // shorter string means that uri was encoded
+    if (res == HTP_OK && prevlen > bstr_len(tx->parsed_uri->path)) {
+        HtpTxUserData *htud = (HtpTxUserData *) htp_tx_get_user_data(tx);
```

Comment:
there is a lot (100%) of duplication here, can we avoid that?

Reply:
Good idea